components/base/wheeled/wheeled_base.go

```diff
@@ -396,3 +403,73 @@ func CreateWheeledBase(
 
 	return base, nil
 }
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+func (base *wheeledBase) WrapWithKinematics(slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel(base.cfg)
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) buildModel(cfg config.Component) (referenceframe.Model, error) {
+	// TODO(rb): this is a hacky workaround for not having kinematics for bases yet
+	// we create a sphere that would encompass the config geometry's rotation a full 360 degrees
+	// determine offset to use
+	geoCfg := cfg.Frame.Geometry
+	r := geoCfg.TranslationOffset.Norm()
+	switch geoCfg.Type {
+	case spatialmath.BoxType:
+		r += r3.Vector{X: geoCfg.X, Y: geoCfg.Y, Z: geoCfg.Z}.Norm()
```

Comment:
You know I saw that thread and was pretty surprised we decided to do that.

Reply:
Thought about this more, I don't think the model needs to be changed here since motion planning doesn't care about what the base thinks is forwards. Everything in this PR is agnostic to how forwards is defined.  But we'll make sure to respect this convention if there is anything that would care about it

Reply:
I think the only thing you have to watch out for in the future is what each dimension of a box or capsule geometry will hit for a collision. Which is why I commented on this line in particular. I feel that you can ignore it now if you use a sphere. Just please add a TODO. There is an RDK ticket that can be linked and added to both teams boards.

Reply:
Just added a TODO calling this out